jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6206 @@
+     * The set operation has the memory semantics of set.
+     *
+	 * @param obj object in which to store the value
+	 * @param offset specifies where to compare and store value in obj
+	 * @param compareValue value that is expected to be in obj at offset
+	 * @param exchangeValue value that will be set in obj at offset if compare is successful
+	 * @return value of the field before this operation. This will be compareValue if the exchange was successful
+     */
+	public final byte compareAndExchangeByte(Object obj, long offset, byte compareValue, byte exchangeValue) {
+		return compareAndExchange8bits(obj, offset, compareValue, exchangeValue);
+	}
+
+    /**
+     * Atomically sets the parameter value at offset in obj if the specified compare value 
+     * matches the existing value in the object.
+     * The get operation has memory semantics of get.
+     * The set operation has the memory semantics of set.
+     * 
+     * @param obj object in which to store the value
+     * @param offset specifies where to compare and store value in obj
+     * @param compareValue value that is expected to be in obj at offset
+     * @param setValue value that will be set in obj at offset if compare is successful
+     * @return boolean value indicating whether the field was updated
+     */
+	public final boolean compareAndSetByte(Object obj, long offset, byte compareValue, byte setValue) {
+		byte exchangedValue = compareAndExchangeByte(obj, offset, compareValue, setValue);
+		return (compareValue == exchangedValue);
+	}
+
+    /**
+     * Sets the parameter value at offset in obj if the specified compare value 
+     * matches the existing value in the object.
+     * The get operation has memory semantics of get.
+     * The set operation has the memory semantics of set.
+     * 
+     * @param obj object in which to store the value
+     * @param offset specifies where to compare and store value in obj
+     * @param compareValue value that is expected to be in obj at offset
+     * @param setValue value that will be set in obj at offset if compare is successful
+     * @return boolean value indicating whether the field was updated
+     */
+	public final boolean weakCompareAndSetByte(Object obj, long offset, byte compareValue, byte setValue) {
+		return compareAndSetByte(obj, offset, compareValue, setValue);
+	}
+
+    /**
+     * Sets the parameter value at offset in obj if the specified compare value 
+     * matches the existing value in the object.
+     * The get operation has memory semantics of getAcquire.
+     * The set operation has the memory semantics of set.
+     * 
+     * @param obj object in which to store the value
+     * @param offset specifies where to compare and store value in obj
+     * @param compareValue value that is expected to be in obj at offset
+     * @param setValue value that will be set in obj at offset if compare is successful
+     * @return boolean value indicatin
```

Comment:
It's marginally more efficient if you inline the call into the if statement (avoids storing and loading a local).